### PR TITLE
fix: add examples for commands

### LINKED ISSUE #378: Add examples for commands
**Describe the solution you'd like**
Add examples for  test, publish, config commands. Eventually, we’d want the examples  to show all the possible ways to use the command. Ultimately this will look something this(`kubectl`): 
image

**Required knowledge**
Golang basic level.

”**How to implement” Suggestion**
In a nutshell, we’ll use Cobra’s `Command` . For instance, adding examples for test command will look like this:

```
cmd := &cobra.Command{
	Use: "name of command",
	Example: Examples(`
		# test the configuration located in root kube folder
		datree test ./kube/**/**.yaml`)
	Run: func(cmd *cobra.Command, args []string) {...},
}
```
1. Inside `pkg/utils`, Add `Examples()` function that accept an input string and will normalize the input. For that, add also a private struct  `normalizer` that will handle the normalization. 

> ``` 
> // Examples normalizes a command's examples.
> func Examples(s string) string {
> 	if len(s) == 0 {
> 		return s
> 	}
> 	return normalizer{s}.trim().indent().string
> }
> 
> // normalizer is a helper struct to normalize the output of the examples.
> const Indentation = `  `
> 
> type normalizer struct {
> 	string
> }
> 
> func (s normalizer) trim() normalizer {
> 	s.string = strings.TrimSpace(s.string)
> 	return s
> }
> 
> func (s normalizer) indent() normalizer {
> 	indentedLines := []string{}
> 	for _, line := range strings.Split(s.string, "\n") {
> 		trimmed := strings.TrimSpace(line)
> 		indented := Indentation + trimmed
> 		indentedLines = append(indentedLines, indented)
> 	}
> 	s.string = strings.Join(indentedLines, "\n")
> 	return s
> }
> ```

2. Create and use the examples in all commands: publish `/cmd/publish/main.go`, test `/cmd/test/main.go`, and config `/cmd/config/main.go`

### PR DESCRIPTION
**Pull request by shubham4443**
Fixes #378 

Please let me know if you want to display examples other than this or need changes in these ones.

### CODE REVIEW

File: cmd/publish/main.go [hunk 1]
```diff
@@ -39,6 +40,13 @@ func New(ctx *PublishCommandContext) *cobra.Command {
 		Use:   "publish <fileName>",
 		Short: "Publish policies configuration for given <fileName>.",
 		Long:  "Publish policies configuration for given <fileName>. Input should be the path to the Policy-as-Code yaml configuration file",
+		Example: utils.Example(`
+		# Publish the policies configuration YAML file
+		datree publish policies.yaml
+
+		# Note
+		You need to first enable Policy-as-Code (PaC) on the settings page in the dashboard
```
Comment (myishay) [inline cmd/publish/main.go]: I would out this one on the same line as the note (so it would also be commented-out)

File: cmd/config/main.go [hunk 2]
```diff
@@ -31,6 +32,10 @@ func New(ctx *ConfigCommandContext) *cobra.Command {
 		Use:   "config",
 		Short: "Configuration management",
 		Long:  `Internal configuration management for datree config file`,
+		Example: utils.Example(`
+		# Change the token in the datree config.yaml file
+		datree config set token MY_EXAMPLE_TOKEN
```
Comment (myishay) [inline cmd/config/main.go]: Would be better adding `<>` to the token example for better understanding

File: cmd/test/main.go [hunk 3]
```diff
@@ -79,6 +80,13 @@ func New(ctx *TestCommandContext) *cobra.Command {
 		Use:   "test <pattern>",
 		Short: "Execute static analysis for given <pattern>",
 		Long:  "Execute static analysis for given <pattern>. Input should be glob or `-` for stdin",
+		Example: utils.Example(`
```
Comment (myishay) [inline cmd/test/main.go:104]: There should also be reference here to sending manifests to datree through stdin.
for example: `cat kube-prod/deployment.yaml | datree test -`

**Review by myishay — COMMENTED** [review]
@shubham4443 Thank you for your contribution!

All in all this PR looks really good!
I added a few minor comments, lmk what you think.

**Comment by shubham4443** [comment]
@myishay Thank you for your response!

I've made the changes as per your request. Please let me know if there is anything more.

Thanks!

**Comment by myishay** [comment]
Nice work!
I will update your branch with the latest changes and we are good to go

**Review by myishay — APPROVED** [review]

**Pull request merged**